src/Pos/Wallet/Web/Server/Full/Common.hs

```diff
@@ -70,40 +59,29 @@ convertHandler
     -> DelegationVar
     -> PeerStateSnapshot
     -> ConnectionsVar
-    -> WebHandler a
+    -> WalletWebMode a
     -> Handler a
 convertHandler nc modernDBs tlw ssc ws delWrap psCtx
                conn handler =
-    liftIO (realRunner . walletRunner $ handler) `Catch.catches` excHandlers
+    liftIO (walletRunner handler) `Catch.catches` excHandlers
   where
-    walletRunner = runWalletWebDB ws
-      . runWalletWS conn
-      . runWalletRedirects
 
-    realRunner :: forall t . RealMode WalletSscType t -> IO t
-    realRunner (RealMode act) = runProduction
-           . usingLoggerName "wallet-api"
-           . runWithoutJsonLogT
-           . flip Ether.runReadersT nc
-           . (\m -> do
-               peerStateCtx <- peerStateFromSnapshot psCtx
-               Ether.runReadersT m
-                   ( Tagged @NodeDBs modernDBs
-                   , Tagged @SscMemTag ssc
-                   , Tagged @TxpHolderTag tlw
-                   , Tagged @DelegationVar delWrap
-                   , Tagged @PeerStateTag peerStateCtx
-                   ))
-           . runDBPureRedirect
-           . runBlockDBRedirect
-           . runSlotsDataRedirect
-           . runSlotsRedirect
-           . runDiscoveryRedirect
-           . runPeerStateRedirect
-           . runGStateCoreRedirect
-           . runBListenerStub
-           $ act
+    walletRunner :: forall a . WalletWebMode a -> IO a
+    walletRunner act = runProduction $ do
+        peerStateCtx <- peerStateFromSnapshot psCtx
+        Mtl.runReaderT (unWalletWebMode act) $
+            WalletWebModeContext
+                ws
+                conn
+                (RealModeContext
```

Comment:
I suppose it should become unnecessary to pass so much stuff to this function and recreate already existing `RealNodeContext`. Probably it's not done because it's WIP.